flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClientFactory.java

```diff
@@ -77,6 +78,8 @@ public ApplicationId getClusterId(Configuration configuration) {
     private YarnClusterDescriptor getClusterDescriptor(Configuration configuration) {
         final YarnClient yarnClient = YarnClient.createYarnClient();
         final YarnConfiguration yarnConfiguration = new YarnConfiguration();
+        yarnConfiguration.addResource(HadoopUtils.getHadoopConfiguration(configuration));
+        yarnConfiguration.addResource(Utils.getYarnConfiguration(configuration));
```

Comment:
Would it make sense to load the `YarnConfiguration` through a utility. Maybe even `HadoopUtils`? That way it would always be consistent.

Reply:
If `HadoopUtils` provides both `getHadoopConfiguration` and `getYarnConfiguration`, then it could be consistent and simplify code https://github.com/apache/flink/pull/14477#discussion_r553198430.
But I am not sure if it is really appropriate and standardized to put `getYarnConfiguration` in the `flink-hadoop-fs` module, because the `flink-yarn` module itself has `Utils`, it may be more clear to use it directly.

Reply:
Yes, it is better to put it into the `Utils` class of `flink-yarn`.